x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -658,10 +678,26 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
             }
         } catch (Exception e) {
             if (ignoreMalformed) {
-                if (subParser != null) {
-                    // close the subParser so we advance to the end of the object
+                if (malformedContentForSyntheticSource != null) {
+                    if (subParser != null) {
```

Comment:
I'm trying to reason about this if statement. I think `subParser` can only be not null here if some kind of error has happened advancing the parser or constructing `XContentSubParser` So this is kind of rare, but we need to take it into account and gather the parser events for ignored source. In the code currently in main, we only care about avoiding NPE when closing the `subParser`.  Does this sound about right?

Reply:
Yes, that's correct but It's not that rare. `subParser` will be null every time when field value is not an object, we hit this code path in tests quite often.

Reply:
Gotcha, I forgot about the validation that happens just before the subParser variable gets assigned. I guess I was referring to when we actually get valid content.